### DISCUSSION TITLE
Search word under cursor

### DISCUSSION THREAD

**Discussion by gabrielbidula** [JetBrains/ideavim#326] [Ideas]
Hey,

Does anyone know if it is already possible to do a **search under the cursor** and **display the results in the find in files window**?
I do this in Neovim by combining Ag(silver searcher) and FZF.

```
" FZF search word under cursor with AG
nnoremap <silent> <Leader>ag :Ag <C-R><C-W><CR>
```

Thanks

**Comment by AlexPl292** [comment]
Hi! Unfortunately, the `Ag` plugin isn't supported by IdeaVim.

**Reply by gabrielbidula** [reply]
Hi, I see, but there is no other way to search under the cursor via keybind? I mean, doesn't need to be with an external plugin it could be done just like Find in files works but passing the word when the keymap is pressed.

**Reply by tmontaigu** [reply]
something like this would do it:

```
nnoremap <silent> <Leader>ag viw:action FindInPath<CR>
```

**Reply by gabrielbidula** [reply]
That works but if I press ESC on the find window I return on visual mode, can we avoid that and return to normal?

**Reply by tmontaigu** [reply]
I don't know if that is possible, or at least I don't know how

**Reply by jackpnm** [reply]
> That works but if I press ESC on the find window I return on visual mode, can we avoid that and return to normal?

This is working for me over the last 10 minutes:
```
map <silent> <Leader>ag viw:action FindInPath<CR><Esc>
```
